packages/dds/tree/src/core/edit-manager/editManager.ts

```diff
@@ -224,18 +224,47 @@ export class EditManager<
         commitToRebase: Commit<TChangeset>,
         branch: Branch<TChangeset>,
     ): TChangeset {
-        if (!branch.isDivergent && commitToRebase.sessionId === this.getLastCommit()?.sessionId) {
+        const peerSessionId = commitToRebase.sessionId;
+        if (!branch.isDivergent && peerSessionId === this.getLastCommit()?.sessionId) {
             // The new commit is not divergent and therefore doesn't need to be rebased.
             return commitToRebase.changeset;
         }
 
-        const changeRebasedToRef = branch.localChanges.reduceRight(
+        const trunkCommits = this.getCommitsAfter(branch.refSeq);
+
+        // The peer branch (or this last edit) is divergent, meaning some edits on the trunk after `branch.refSeq`
```

Comment:
I think there is a simpler approach, which would also allow us to remove the `isDivergent` logic. In `addSequencedChange` we could skip adding the change to a branch if the change's ref number refers to the most recent change in the trunk. Does that seem correct?